skimage/filters/_gaussian.py

```diff
@@ -113,8 +117,17 @@ def gaussian(image, sigma=1, output=None, mode='nearest', cval=0,
         if len(sigma) != image.ndim:
             sigma = np.concatenate((np.asarray(sigma), [0]))
     image = convert_to_float(image, preserve_range)
-    return ndi.gaussian_filter(image, sigma, mode=mode, cval=cval,
-                               truncate=truncate)
+    if output is None:
+        output = np.empty_like(image)
+    elif not isinstance(output, np.ndarray):
```

Comment:
I don't think this is quite right.

I kinda like making the assumption of `array-like` interfaces. 

Output is only used when users want to save on memory allocation and deallocation. they should know it is a numpy array, and not a list.

I would do
https://docs.scipy.org/doc/numpy/reference/generated/numpy.issubdtype.html

```
output is not None and np.issubdtype(output.dtype, np.floating)
```
and let scipy deal with cross floating point precision arithmetic.

Reply:
If user know that, the `output` is numpy array and this information is well documented, then the suggestion is pretty good.

Reply:
The reason i suggest against this is that many people are experimentng with numpy-like arrays.

GPU arrays, distributed arrays, they are becoming fashionable, and pretty useful!

Reply:
You are right, it is very useful comment. I think in this case definitely no need to limit the output type.